tensorboard/webapp/metrics/store/metrics_store_internal_utils_test.ts

```diff
@@ -646,6 +647,173 @@ describe('metrics store utils', () => {
+      [imageCardId]: {
+        runId: 'test run Id',
+        plugin: PluginType.IMAGES,
+        tag: 'tagC',
+        sample: 111,
+      },
+    };
+    let timeSeriesData: TimeSeriesData = {
+      scalars: {},
+      histograms: {},
+      images: {},
+    };
+
+    beforeEach(() => {
+      timeSeriesData = {
+        scalars: {},
+        histograms: {},
+        images: {
+          tagC: {
+            111: {
+              runToLoadState: {},
+              runToSeries: {
+                'test run Id': [
+                  {step: 10, wallTime: 0, imageId: '1'},
+                  {step: 20, wallTime: 10, imageId: '2'},
+                  {step: 30, wallTime: 15, imageId: '3'},
+                ],
+              },
+            },
+          },
+        },
+      };
+    });
+
+    it(`updates cardStepIndex to matched selected time`, () => {
+      const selectedTime = {start: {step: 20}, end: null};
+      const nextCardStepIndex = generateNextCardStepIndexFromSelectedTime(
+        previousCardStepIndex,
+        cardMetadataMap,
+        timeSeriesData,
+        selectedTime
+      );
+
+      expect(nextCardStepIndex).toEqual({[imageCardId]: 1});
+    });
+
+    it(`does not update cardStepIndex on other non-image plugin`, () => {
+      const histogramCardId = 'test histogram card id "plugin":"histogram"';
+      const previousCardStepIndexWtihHistogram = {
+        [histogramCardId]: null,
+      };
+      const cardMetadataMapWtihHistogram = {
+        [histogramCardId]: {
+          runId: 'test run Id',
+          plugin: PluginType.HISTOGRAMS,
+          tag: 'tagB',
+          sample: 111,
+        },
+      };
+      const timeSeriesDataWtihHistogram = {
+        scalars: {},
+        histograms: {
+          tagB: {
+            runToLoadState: {},
+            runToSeries: {
+              'test run Id': [
+                {step: 10, wallTime: 0, bins: [{min: 0, max: 10, count: 2}]},
+                {step: 20, wallTime: 10, bins: [{min: 0, max: 10, count: 2}]},
+                {step: 30, wallTime: 15, bins: [{min: 0, max: 10, count: 2}]},
+              ],
+            },
+          },
+        },
+        images: {},
+      };
+      const selectedTime = {start: {step: 20}, end: null};
+
+      const nextCardStepIndex = generateNextCardStepIndexFromSelectedTime(
+        previousCardStepIndexWtihHistogram,
+        cardMetadataMapWtihHistogram,
+        timeSeriesDataWtihHistogram,
+        selectedTime
+      );
+      expect(nextCardStepIndex).toEqual({
+        [histogramCardId]: null,
+      });
+    });
+
+    it(`does not update cardStepIndex on selected step with no image`, () => {
+      const selectedTime = {start: {step: 15}, end: null};
+      const nextCardStepIndex = generateNextCardStepIndexFromSelectedTime(
+        previousCardStepIndex,
+        cardMetadataMap,
+        timeSeriesData,
+        selectedTime
+      );
+
+      expect(nextCardStepIndex).toEqual({[imageCardId]: null});
```

Comment:
I'd find this test much more reassuring if the original value for imageCardId were anything other than null. Otherwise, I might believe that you are taking the null value from getNextImageCardStepIndexFromSingleSelection and using it.